gapis/gfxapi/gles/resources.go

```diff
@@ -203,17 +204,28 @@ func (shader *Shader) SetResourceData(ctx context.Context, at *path.Command,
 	for resource.Accesses[index] > at.Index && index >= 0 {
 		index--
 	}
+
+	to_mutate := len(list.Atoms)
+	if to_mutate > 5 {
+		to_mutate = 5
+	}
+	// Hack until the architecture is part of the header
```

Comment:
I will wait and then rebase this on top of your change. I was unsure when that CL would make it up, and I would rather not make you rebase on top of these changes either.